flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/TableEnvHiveConnectorTest.java

```diff
@@ -138,6 +138,34 @@ private void readWriteFormat(String format) throws Exception {
 		hiveShell.execute("drop database db1 cascade");
 	}
 
+	@Test
+	public void testDecimal() throws Exception {
+		hiveShell.execute("create database db1");
+		try {
+			// Hive's default decimal is decimal(10, 0)
+			hiveShell.execute("create table db1.src1 (x decimal)");
+			hiveShell.execute("create table db1.src2 (x decimal)");
+			hiveShell.execute("create table db1.dest (x decimal)");
+			// populate src1 from Hive
+			hiveShell.execute("insert into db1.src1 values (1),(2.0),(5.4),(5.5),(123456789123)");
+
+			TableEnvironment tableEnv = getTableEnvWithHiveCatalog();
+			// populate src2 with same data from Flink
+			tableEnv.sqlUpdate("insert into db1.src2 values (cast(1 as decimal(10,0))), (cast(2.0 as decimal(10,0))), " +
```

Comment:
could we have some non-zero scale decimal types?

Reply:
Do you mean insert some non-zero-scale decimals into zero-scale decimal column? It's not allowed by the planner.

Reply:
no. I meant testing decimals with non-zero scales, which is more general. From what I saw, all the decimals have zero scales.